Source/AwsCommonRuntimeKit/mqtt/MqttRequestResponseClient.swift

```diff
@@ -181,60 +220,154 @@ public struct RequestResponseOperationOptions: CStructWithUserData, Sendable {
     }
 }
 
+private func MqttRRStreamingOperationTerminationCallback(_ userData: UnsafeMutableRawPointer?) {
+    // Termination callback. This is triggered when the native object is terminated.
+    // It is safe to release the native operation at this point. `takeRetainedValue()` would release 
+    // the operation reference. ONLY DO IT AFTER YOU NEED RELEASE THE OBJECT
+    _ = Unmanaged<StreamingOperationCore>.fromOpaque(userData!).takeRetainedValue()
+}
+
+private func MqttRRStreamingOperationIncomingPublishCallback(_ publishEvent: UnsafePointer<aws_mqtt_rr_incoming_publish_event>?,
+                                                             _ userData: UnsafeMutableRawPointer?) {
+    guard let userData, let publishEvent else {
+        // No userData, directly return
+        return
+    }
+    let operationCore = Unmanaged<StreamingOperationCore>.fromOpaque(userData).takeUnretainedValue()
+    operationCore.rwlock.read {
+        // Only invoke the callback if the streaming operation is not closed.
+        if operationCore.rawValue != nil, operationCore.options.incomingPublishEventHandler != nil {
+            let subStatusEvent = IncomingPublishEvent(publishEvent)
+            operationCore.options.incomingPublishEventHandler!(subStatusEvent)
+        }
+    }
+}
+
+private func MqttRRStreamingOperationSubscriptionStatusCallback(_ eventType: aws_rr_streaming_subscription_event_type,
+                                                                _ errorCode: Int32,
+                                                                _ userData: UnsafeMutableRawPointer?) {
+    guard let userData else {
+        // No userData, directly return
+        return
+    }
+    let operationCore = Unmanaged<StreamingOperationCore>.fromOpaque(userData).takeUnretainedValue()
+    operationCore.rwlock.read {
+        // Only invoke the callback if the streaming operation is not closed.
+        if operationCore.rawValue != nil, operationCore.options.subscriptionStatusEventHandler != nil {
+            let subStatusEvent = SubscriptionStatusEvent(event: SubscriptionStatusEventType(eventType),
+                                                         error: errorCode == 0 ? nil : CRTError(code: Int32(errorCode)))
+            operationCore.options.subscriptionStatusEventHandler!(subStatusEvent)
+        }
+    }
+}
+
 /// Configuration options for streaming operations
-public struct StreamingOperationOptions: CStruct, Sendable {
-    let subscriptionStatusEventHandler: SubscriptionStatusEventHandler
-    let incomingPublishEventHandler: IncomingPublishEventHandler
-    let topicFilter: String
+public struct StreamingOperationOptions: CStructWithUserData, Sendable {
+    /// The MQTT topic that the streaming is "listen" to.
+    public let topicFilter: String
+    /// (Optional) The handler function a streaming operation will use for subscription status events.
```

Comment:
Trivial: Our service clients will ALWAYS pass down these callbacks. They should never be nil and if they are, something is broken with the generated service client. We can optionally change this to a non optional property and handle it as such.